# **Apple Apps Data Analysis**

# **1.About the Author** 

Hello! I’m Asadullah Shehbaz — an aspiring Data Scientist on a journey of discovery and growth in the world of data.
I'm passionately exploring the realms of data science, diving deep into analytical techniques, machine learning, and real-world problem-solving to sharpen my skills and contribute meaningfully to this dynamic field.

> "I believe that community is the cornerstone of growth and innovation — by learning together, we can go further, faster, and stronger".

| Name               | Email                                               | LinkedIn                                                  | GitHub                                           | Kaggle                                        |
|--------------------|-----------------------------------------------------|-----------------------------------------------------------|--------------------------------------------------|-----------------------------------------------|
| **Asadullah Shehbaz**      |asadullahcreative@gmail.com  | [![LinkedIn Badge](https://img.shields.io/badge/LinkedIn-%23000000.svg?style=for-the-badge&logo=LinkedIn&logoColor=white)](https://www.linkedin.com/in/asadullah-shehbaz-18172a2bb/)  | [![GitHub Badge](https://img.shields.io/badge/GitHub-%23000000.svg?style=for-the-badge&logo=GitHub&logoColor=white)](https://github.com/AsadullahShehbaz)  | [![Kaggle Badge](https://img.shields.io/badge/Kaggle-%23000000.svg?style=for-the-badge&logo=Kaggle&logoColor=white)](https://www.kaggle.com/asadullahcreative)  |

## **2.About the Data** 

- *This Data Contains **21** Column / Feature / Attribute / variables*  
- *This Data Contains **1230376** Rows / Data Point / Record / Entry / Data Row*  
- *You Can Get This Data From* [HERE](https://www.kaggle.com/datasets/gauthamp10/apple-appstore-apps)




| #   | Name                          | Define                                             |
| --- |------------------------------------|----------------------------------------------------|
| 1   | App_Id                             | Unique identifier for the Apple app.              |
| 2   | App_Name                           | The name or title of the Apple app.               |
| 3   | AppStore_Url                       | The URL of the app's page on the App Store.       |
| 4   | Primary_Genre                      | The primary genre or category of the app.        |
| 5   | Content_Rating                     | The age rating or content restrictions for the app. |
| 6   | Size_Bytes                         | The size of the app in bytes.                     |
| 7   | Required_IOS_Version               | The minimum iOS version required to run the app.  |
| 8   | Released                           | The date when the app was originally released.    |
| 9   | Updated                            | The date of the last update to the app.           |
| 10  | Version                            | The current version of the app.                   |
| 11  | Price                              | The price of the app (if not free).               |
| 12  | Currency                           | The currency used for pricing the app.            |
| 13  | Free                               | Indicates whether the app is free .       |
| 14  | DeveloperId                        | Unique identifier for the app's developer.        |
| 15  | Developer                          | The name of the app's developer.                  |
| 16  | Developer_Url                      | The URL of the developer's page on the App Store. |
| 17  | Developer_Website                  | The official website of the app's developer.     |
| 18  | Average_User_Rating                | The average user rating of the app.              |
| 19  | Reviews                            | The number of user reviews for the app.          |
| 20  | Current_Version_Score              | The user rating for the current app version.     |
| 21  | Current_Version_Reviews            | The number of reviews for the current app version. |










---
## **3. Task**

<!-- reread day 19 notes and add your questions and and answers here 
 -->
- EDA helps us find patterns in data, similar to discovering a treasure map.
- It guides us in selecting the right tools and methods, like choosing gear for an adventure.
- EDA uncovers valuable insights, just like to finding secret clues for informed decisions.
- We document observations during the coding process.
- Ultimately, EDA leads to a summary and conclusions from our findings.

---
## **4. Objective**

- Goal: Analyze the dataset thoroughly.
- Purpose: Understand customer preferences and needs.
- Benefit: Share insights with developers and stakeholders.
- Outcome: Improve upcoming applications to attract more business.

---
## **5. Analysis Software Tools Used**


Names | Versions | Their Use |
---------|----------| --------|
 Python | 3.10.0 | Python is a versatile programming language widely used in data science for its simplicity, readability, and extensive libraries. |
 Pandas | 1.5.3  |Pandas is a powerful data manipulation and analysis library in Python, providing data structures and functions to efficiently work with structured data. |
 Matplotlib | 3.6.2 | Matplotlib is a plotting library in Python that enables the creation of various types of static, animated, and interactive visualizations.|
 Numpy | 1.23.5 | NumPy is a fundamental library for numerical computing in Python, providing support for large, multi-dimensional arrays and a collection of mathematical functions to operate on these arrays efficiently. |
 Seaborn | 0.12.2 | Seaborn is a statistical data visualization library built on top of Matplotlib, offering a higher-level interface for creating attractive and informative statistical graphics.|

---
## **6. Data Cleaning**



In [3]:
# import libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline
# %matplotlib inline is used to display plots directly within the jupyter notebook 

#### **Knowing The Data**

In [4]:
# Imputing the dataset using pandas 
df = pd.read_csv('../data/appleAppData.csv')

In [5]:
# By using these commands , we can see full display of columns and rows data 
pd.set_option('display.max.columns',None)
pd.set_option('display.max.rows',None)

In [6]:
df.head()

,App_Id,App_Name,AppStore_Url,Primary_Genre,Content_Rating,Size_Bytes,Required_IOS_Version,Released,Updated,Version,Price,Currency,Free,DeveloperId,Developer,Developer_Url,Developer_Website,Average_User_Rating,Reviews,Current_Version_Score,Current_Version_Reviews
0,com.hkbu.arc.apaper,A+ Paper Guide,https://apps.apple.com/us/app/a-paper-guide/id...,Education,4+,21993472.0,8.0,2017-09-28T03:02:41Z,2018-12-21T21:30:36Z,1.1.2,0.00,USD,True,1375410542,HKBU ARC,https://apps.apple.com/us/developer/hkbu-arc/i...,NaN,0.0,0,0.0,0
1,com.dmitriev.abooks,A-Books,https://apps.apple.com/us/app/a-books/id103157...,Book,4+,13135872.0,10.0,2015-08-31T19:31:32Z,2019-07-23T20:31:09Z,1.3,0.00,USD,True,1031572001,Roman Dmitriev,https://apps.apple.com/us/developer/roman-dmit...,NaN,5.0,1,5.0,1
2,no.terp.abooks,A-books,https://apps.apple.com/us/app/a-books/id145702...,Book,4+,21943296.0,9.0,2021-04-14T07:00:00Z,2021-05-30T21:08:54Z,1.3.1,0.00,USD,True,1457024163,Terp AS,https://apps.apple.com/us/developer/terp-as/id...,NaN,0.0,0,0.0,0
3,fr.antoinettefleur.Book1,A-F Book #1,https://apps.apple.com/us/app/a-f-book-1/id500...,Book,4+,81851392.0,8.0,2012-02-10T03:40:07Z,2019-10-29T12:40:37Z,1.2,2.99,USD,False,439568839,i-editeur.com,https://apps.apple.com/us/developer/i-editeur-...,NaN,0.0,0,0.0,0
4,com.imonstersoft.azdictionaryios,A-Z Synonyms Dictionary,https://apps.apple.com/us/app/a-z-synonyms-dic...,Reference,4+,64692224.0,9.0,2020-12-16T08:00:00Z,2020-12-18T21:36:11Z,1.0.1,0.00,USD,True,656731821,Ngov chiheang,https://apps.apple.com/us/developer/ngov-chihe...,http://imonstersoft.com,0.0,0,0.0,0


In [7]:
# Let's take a quick overview of data 
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 1230376 entries, 0 to 1230375
Data columns (total 21 columns):
 #   Column                   Non-Null Count    Dtype  
---  ------                   --------------    -----  
 0   App_Id                   1230376 non-null  object 
 1   App_Name                 1230375 non-null  object 
 2   AppStore_Url             1230376 non-null  object 
 3   Primary_Genre            1230376 non-null  object 
 4   Content_Rating           1230376 non-null  object 
 5   Size_Bytes               1230152 non-null  float64
 6   Required_IOS_Version     1230376 non-null  object 
 7   Released                 1230373 non-null  object 
 8   Updated                  1230376 non-null  object 
 9   Version                  1230376 non-null  object 
 10  Price                    1229886 non-null  float64
 11  Currency                 1230376 non-null  object 
 12  Free                     1230376 non-null  bool   
 13  DeveloperId              1230376 non-null 

_**Observations**_ No 1  :
-  We have bool(1), float64(4), int64(3), object(13)
- Rows : 1230376 entries, 0 to 1230375
- Data columns : total 21 columns

In [8]:
df.describe()

,Size_Bytes,Price,DeveloperId,Average_User_Rating,Reviews,Current_Version_Score,Current_Version_Reviews
count,1.230152e+06,1.229886e+06,1.230376e+06,1.230376e+06,1.230376e+06,1.230376e+06,1.230376e+06
mean,7.510857e+07,5.041264e-01,1.092869e+09,1.788387e+00,8.259478e+02,1.788387e+00,8.259478e+02
std,2.318726e+08,6.869674e+00,4.090897e+08,2.154105e+00,5.016051e+04,2.154105e+00,5.016051e+04
min,2.762800e+04,0.000000e+00,2.816565e+08,0.000000e+00,0.000000e+00,0.000000e+00,0.000000e+00
25%,2.041344e+07,0.000000e+00,7.381109e+08,0.000000e+00,0.000000e+00,0.000000e+00,0.000000e+00
50%,4.333158e+07,0.000000e+00,1.179711e+09,0.000000e+00,0.000000e+00,0.000000e+00,0.000000e+00
75%,8.365466e+07,0.000000e+00,1.471960e+09,4.333330e+00,3.000000e+00,4.333330e+00,3.000000e+00
max,7.677945e+10,9.999900e+02,1.590233e+09,5.000000e+00,2.268533e+07,5.000000e+00,2.268533e+07


***Observations No 2***
- We can see count , mean , std , min , and other insights but we did not clean the data so these insights can be different after cleaning the data dealing with null values etc 

### **Lets Handle Missing Values**

In [11]:
df.isnull().sum().sort_values(ascending=False)

Developer_Website          643988
Developer_Url                1109
Price                         490
Size_Bytes                    224
Released                        3
App_Name                        1
Primary_Genre                   0
AppStore_Url                    0
App_Id                          0
Content_Rating                  0
Version                         0
Updated                         0
Required_IOS_Version            0
Free                            0
Currency                        0
Developer                       0
DeveloperId                     0
Average_User_Rating             0
Reviews                         0
Current_Version_Score           0
Current_Version_Reviews         0
dtype: int64


Column | Missing_Values  | Dealing Method |
---------|---------- | ---------- |
 Size_Bytes |  224   | Replace with the mean of data |
 Price | 490  | Replace with the mean of data |
 Developer_Url | 1109  | As the rows contain other information so replace null values with "No_Url" |
 App_Name | 1  | Remove that Column |
 Released | 3 | Replace with mean of date time |

## "Let's reach our initial goal of Dealing with Null Values"


In [18]:
# percentage of missing values 
df.isnull().sum().sort_values(ascending=False)*100/len(df)

Developer_Website          52.340748
Developer_Url               0.090135
Price                       0.039825
Size_Bytes                  0.018206
Released                    0.000244
App_Name                    0.000081
Primary_Genre               0.000000
AppStore_Url                0.000000
App_Id                      0.000000
Content_Rating              0.000000
Version                     0.000000
Updated                     0.000000
Required_IOS_Version        0.000000
Free                        0.000000
Currency                    0.000000
Developer                   0.000000
DeveloperId                 0.000000
Average_User_Rating         0.000000
Reviews                     0.000000
Current_Version_Score       0.000000
Current_Version_Reviews     0.000000
dtype: float64

```Dealing with Developer_Website Column``` 

- Here we got a column **[Developer_Website]** which contain 52 percent of missing values 
- Observations : we can say that More then 50% developers dont share or provide 
their develepor website while uploading app on apple app playstore 
  